### PR TITLE
An assortment of small fixes and optimizations

### PR DESCRIPTION
**Pull request by wbinventor**
This short PR adds an assortment of small fixes and optimizations that I've made to make it possible to model the fully detailed 2D BEAVRS core with 10,000s of materials. These include:

* Improve performance of `load_openmc_mgxs_lib(...)` by caching the number of materials in the geometry
* Improve performance of `get_openmoc_geometry(...)` from an OpenCG geometry by eliminating compatibility checks unless necessary
* Optimized and parallelized CMFD centroid generation
* Added a new `Material::copy(Material* material)` method to make it easy to design algorithms to populate large geometries with lots of different materials
* Added an optional user-defined track filename suffix to make it easy to save/load track files for different geometries without them being overwritten

### CODE REVIEW

File: openmoc/opencg_compatible.py [hunk 1]
```diff
@@ -1014,9 +1018,6 @@ def get_openmoc_geometry(opencg_geometry):
     opencg_geometry.assign_auto_ids()
     opencg_geometry = copy.deepcopy(opencg_geometry)
 
-    # Update Cell bounding boxes in Geometry
-    opencg_geometry.update_bounding_boxes()
```
Comment (samuelshaner) [inline openmoc/opencg_compatible.py]: Why was this removed?

Comment (wbinventor) [inline openmoc/opencg_compatible.py]: To be honest, I don't recall why I even had it in here, and I don't see any reason to perform this calculation before building an OpenMOC geometry. It was creating a bottleneck for a version of the BEAVRS core with over 50,000+ each of materials, cells and universes that I didn't want to wait on, so I removed it. Do you need it?

File: src/Material.cpp [hunk 2]
```diff
@@ -1039,6 +1039,37 @@ Material* Material::clone() {
 
 
 /**
+ * @brief Copy another Material's cross section data into this Material.
+ * @details This method performs a deep copy of the other Material's
+ *          cross section data arrays for use by this Material.
+ * @param material a pointer to the Material with data to copy
+ */
+void Material::copy(Material* material) {
```
Comment (samuelshaner) [inline src/Material.cpp]: Since this method doesn't return a copy (what one might expect from a copy method) but instead copies *from* the input material to the current material, I would recommend naming this method `copyFrom(...)`?

**Comment by wbinventor** [comment]
By the way, I removed the optimizations / parallelization that I put into `Cmfd::generateKNearest()` since I discovered that they resulted in a segmentation fault when compiled in double precision.

File: openmoc/opencg_compatible.py (hunk 1 above)
Comment (samuelshaner) [inline openmoc/opencg_compatible.py]: Nope, just curious if it was removed by mistake.

File: src/TrackGenerator.h [hunk 3]
```diff
@@ -79,6 +79,9 @@ class TrackGenerator {
   /** Filename for the *.tracks input / output file */
   std::string _tracks_filename;
 
+  /** Filename suffix for the *.tracks input / output file */
+  std::string _tracks_filename_suffix;
```
Comment (samuelshaner) [inline src/TrackGenerator.h]: Do you just have this so you can put in custom names for track files and run different geometries with the same parameters and not overwrite the tracks file?

**Comment by samuelshaner** [comment]
This looks good to go once the items are addressed.

File: src/TrackGenerator.h (hunk 3 above)
Comment (wbinventor) [inline src/TrackGenerator.h]: Yes - if I want to model the different clustered versions of the same geometry, with the same CMFD mesh and track laydown, I want the track files to be stored separately so I can actually take advantage of them for subsequent simulations. Perhaps even more important, I've noticed issues when running multiple simulations which write/load tracks files at the same time - it can cause one or more simulations to just hang.

File: src/Material.cpp (hunk 2 above)
Comment (wbinventor) [inline src/Material.cpp]: Agreed, I like this name change and just pushed it.
